lib/src/util/mixins.dart

```diff
@@ -265,6 +281,291 @@ mixin InteractiveMixin implements IInteractiveContext, IContextData {
+      await _getInteractionEvent(
+        commands.interactions.events.onButtonEvent,
+        messageIds: [message.id],
+        timeout: timeout,
+        authorOnly: authorOnly,
+      ),
+    );
+
+    context._parent = this;
+    _delegate = context;
+
+    return context;
+  }
+
+  @override
+  Future<T> getButtonSelection<T>(
+    List<T> values,
+    MessageBuilder builder, {
+    Map<T, ButtonStyle>? styles,
+    bool authorOnly = true,
+    ResponseLevel? level,
+    Duration? timeout,
+    FutureOr<ButtonBuilder> Function(T)? toButton,
+    Converter<T>? converterOverride,
+  }) async {
+    if (_delegate != null) {
+      return _delegate!.getButtonSelection(
+        values,
+        builder,
+        authorOnly: authorOnly,
+        converterOverride: converterOverride,
+        level: level,
+        styles: styles,
+        timeout: timeout,
+        toButton: toButton,
+      );
+    }
+
+    assert(
+      toButton == null || converterOverride == null,
+      'Cannot specify both toButton and converterOverride.',
+    );
+
+    toButton ??= converterOverride?.toButton;
+    toButton ??= commands.getConverter(DartType<T>())?.toButton;
+
+    if (toButton == null) {
+      throw UncaughtCommandsException(
+        'No suitable method found for converting $T to ButtonBuilder.',
+        _nearestCommandContext,
+      );
+    }
+
+    Map<String, T> idToValue = {};
+
+    List<ButtonBuilder> buttons = await Future.wait(values.map((value) async {
+      ButtonBuilder builder = await toButton!(value);
+      ButtonStyle? style = styles?[value];
+      String id = createId();
+
+      // We have to copy since the fields on ButtonBuilder are final.
+      return ButtonBuilder(
+        builder.label,
+        id,
+        style ?? builder.style,
+      )
+        ..disabled = builder.disabled
+        ..emoji = builder.emoji;
+    }));
+
+    builder = builderToComponentBuilder(builder);
+
+    while (buttons.isNotEmpty) {
+      // Max 5 buttons per row
+      int lastIndex = min(5, buttons.length);
+
+      ComponentRowBuilder row = ComponentRowBuilder();
+
+      for (final button in buttons.take(lastIndex)) {
+        row.addComponent(button);
+      }
+
+      (builder as ComponentMessageBuilder).addComponentRow(row);
+      buttons.removeRange(0, lastIndex);
+    }
+
+    await respond(builder, level: level);
+
+    ButtonComponentContext context = await commands.contextManager.createButtonComponentContext(
+      await _getInteractionEvent(
+        commands.interactions.events.onButtonEvent,
+        componentIds: idToValue.keys.toList(),
+        timeout: timeout,
+        authorOnly: authorOnly,
+      ),
+    );
+
+    context._parent = this;
+    _delegate = context;
+
+    return idToValue[context.componentId]!;
+  }
+
+  @override
+  Future<bool> getConfirmation(
+    MessageBuilder builder, {
+    Map<bool, String> values = const {true: 'Yes', false: 'No'},
```

Comment:
maybe you should do const somewhere with these values since you are duplicating the same thing multiple times